controllers/managedocs_controller.go

```diff
@@ -911,6 +916,26 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+// reconcilePrometheusService function wait for prometheus Service
+// to start and sets appropriate annotation for 'service-ca' controller
+func (r *ManagedOCSReconciler) reconcilePrometheusService() error {
+	r.Log.Info("Reconciling Prometheus Service")
+	prometheusService := &corev1.Service{}
+	err := r.Client.Get(r.ctx,
+		types.NamespacedName{Name: prometheusServiceName, Namespace: r.namespace},
+		prometheusService)
+	// as this Service is not created explicitly by us,
+	// we don't raise any error, will wait till it is available
+	if err != nil {
+		return nil
+	}
+	// check whether the Service has appropriate annotation
+	if annotVal := prometheusService.Annotations[serviceServingCertAnnotationKeyBeta]; annotVal != prometheusServiceName {
+		prometheusService.Annotations[serviceServingCertAnnotationKeyBeta] = prometheusServiceName
+	}
+	return r.Client.Update(r.ctx, prometheusService)
```

Comment:
We should automatically, this will just cause another requeue to happen (which is not needed and might cause an infinite loop of reconciles) 

Please update only if have to.

Reply:
Yes, changed this part (as mentioned above)...

Reply:
Same question as of the last comment: "Why do we need a secondary service if there is one that already exists? Or am I missing something?"